File: 12.go
package main

// import "fmt"
import "math"

func main() {
    triangular := 28
    incre := 8
    num := 0
    sqrt := 0
    for {
        sqrt = int(math.Sqrt(float64(triangular)))
        for i := 2; i < sqrt; i++ {
            if triangular % i == 0 {
                num++
            }
        }
        if num > 250 {
            break
        }
        num = 0
        triangular += incre
        incre++
    }
    println(triangular)
}

File: 12.c
#include <stdio.h>
#include <math.h>

int main(int argc, char const *argv[]) {
    long triangular = 28;
    unsigned incre = 8;
    unsigned num = 8;
    unsigned sqrted = 0;
    while (1) {
        sqrted = (int)sqrt((double)triangular);
        for (int i = 2; i < sqrted; i++) {
            if (triangular % i == 0) {
                num++;
            }
        }
        if (num > 250) {
            break;
        }
        num = 0;
        triangular += incre;
        incre++;
    }
    printf("%ld\n", triangular);
    return 0;
}